src/barChart.ts

```diff
@@ -407,41 +424,49 @@ export class BarChart implements IVisual {
 
         barSelectionMerged.classed('bar', true);
 
-        const opacity: number = viewModel.settings.generalView.opacity / 100;
+        const opacity: number = this.formattingSettings.generalView.opacity.value / 100;
         barSelectionMerged
             .attr("width", xScale.bandwidth())
             .attr("height", d => height - yScale(<number>d.value))
             .attr("y", d => yScale(<number>d.value))
             .attr("x", d => xScale(d.category))
+            .attr(SubSelectableObjectNameAttribute, 'colorSelector')
+            .attr(SubSelectableDisplayNameAttribute, d => `colorSelector-${d.index}`)
+            .attr(SubSelectableTypeAttribute, powerbi.visuals.SubSelectionStylesType.Shape)
+            .classed(HtmlSubSelectableClass, options.formatMode)
             .style("fill-opacity", opacity)
             .style("stroke-opacity", opacity)
             .style("fill", (dataPoint: BarChartDataPoint) => dataPoint.color)
             .style("stroke", (dataPoint: BarChartDataPoint) => dataPoint.strokeColor)
             .style("stroke-width", (dataPoint: BarChartDataPoint) => `${dataPoint.strokeWidth}px`);
 
         this.tooltipServiceWrapper.addTooltip(barSelectionMerged,
-            (datapoint: BarChartDataPoint) => this.getTooltipData(datapoint),
-            (datapoint: BarChartDataPoint) => datapoint.selectionId
+            (dataPoint: BarChartDataPoint) => this.getTooltipData(dataPoint),
+            (dataPoint: BarChartDataPoint) => dataPoint.selectionId
         );
 
         this.syncSelectionState(
             barSelectionMerged,
             <ISelectionId[]>this.selectionManager.getSelectionIds()
         );
+        if (this.formatMode) {
+            barSelectionMerged.on('click', null);
+            this.svg.on('click', null);
+            this.svg.on('contextmenu', null);
```

Comment:
Let's create a function named hadleFormatModeActions or something similar, so it'll be more obvious why are we doing this.